test/join.jl

```diff
@@ -1557,4 +1559,27 @@ end
                     c="c", d="d")
 end
 
+@testset "matchmissing :notequal correctness" begin
+    name = DataFrame(ID = Union{Int, Missing}[1, 2, missing],
```

Comment:
we should also check:
1. joining on more than 1 column
2. having `missing` in non-key column that is not dropped (to make sure we do not accidentally drop it)
3. location of `missing` not only at the end but also in the front, middle + several scattered `missing`s (just to make sure we cover several different `_findall` cases - generating range or vector)

Thank you!

Reply:
Done